titus-server-master/src/main/java/com/netflix/titus/master/appscale/service/DefaultAppScaleManager.java

```diff
@@ -440,18 +438,17 @@ private String buildAutoScalingGroup(String jobId) {
     static String buildAutoScalingGroupV3(JobDescriptor<?> jobDescriptor) {
         JobGroupInfo jobGroupInfo = jobDescriptor.getJobGroupInfo();
         String jobGroupSequence = jobGroupInfo.getSequence() != null ? jobGroupInfo.getSequence() : DEFAULT_JOB_GROUP_SEQ;
-        List<String> parameterList = Arrays.asList(jobDescriptor.getApplicationName(), jobGroupInfo.getStack(), jobGroupInfo.getDetail(), jobGroupSequence);
-        return buildAutoScalingGroupFromParameters(parameterList);
+        // Using frigga builder for auto scaling group name so that cloud watch alarm configuration
+        // is compatible with spinnaker generated auto scaling group name that is tagged with cloud watch metrics
+        AutoScalingGroupNameBuilder autoScalingGroupNameBuilder = new AutoScalingGroupNameBuilder();
+        String asgWithNoSequence = autoScalingGroupNameBuilder
+                .withAppName(jobDescriptor.getApplicationName())
+                .withStack(jobGroupInfo.getStack())
+                .withDetail(jobGroupInfo.getDetail())
+                .buildGroupName();
+        return String.format("%s-%s", asgWithNoSequence, jobGroupSequence);
```

Comment:
any concerns with this not being necessarily backwards compatible, and will potentially leak old alarms as existing policies are updated?